### ISSUE TITLE
runlevel related coverity issues

### ISSUE THREAD

**Issue by mpreisler** [OpenSCAP/openscap#591]
```
1. Defect type: CHAR_IO 
1. openscap-1.2.12/src/OVAL/probes/unix/runlevel.c:309: char_io: Assigning the return value of "fgetc" to char "c" truncates its value.
Expand
2. Defect type: NULL_RETURNS 
7. openscap-1.2.12/src/OVAL/probes/unix/runlevel.c:307: dereference: Dereferencing a pointer that might be null "osrelease" when calling "fscanf".
```

**Comment by mpreisler** [comment]
```
2. openscap-1.2.12/src/OVAL/probes/unix/runlevel.c:303: dead_error_line: Execution cannot reach the expression "*__errno_location()" inside this statement: "return !*__errno_location();".
```

**Comment by mpreisler** [comment]
```
1. openscap-1.2.12/src/OVAL/probes/unix/runlevel.c:303: null_misuse: Comparing pointer "osrelease" against NULL using anything besides == or != is likely to be incorrect.
```

**Comment by jan-cerny** [comment]
Caused mostly by 7f18f148c84e48ed250f426bb166d72553a15cbd 

We should be more careful when reviewing pull requests. Apart from these issues reported by coverity scan, the patch doesn't follow our coding style.

**Comment by mpreisler** [comment]
Agreed, @jan-cerny . I will try to get this fixed for `1.2.13`.
